cms/static/cms/js/modules/cms.toolbar.js

```diff
@@ -736,7 +736,7 @@ var Toolbar = new Class({
     _refreshMarkup: function(newToolbar) {
         const switcher = this.ui.toolbarSwitcher.detach();
 
-        $(this.ui.toolbar[0]).replaceWith(newToolbar[0]);
+        $(this.ui.toolbar).html(newToolbar.children());
```

Comment:
I like this fix, it would awesome if you add some comments about what was wrong and how does your fix resolves that? Easier to review and refer in future.

Reply:
Thx! Goes back to the analysis I wrote in the issue comments:

There's been a change to `cms.toolbar.js`. Also affected is line 739:

https://github.com/django-cms/django-cms/blob/04e843337d18afe64478d09f00051ed13a49ff55/cms/static/cms/js/modules/cms.toolbar.js#L736-L740

Hypthesis: The DOM is changed without updating the reference in `this.ui.toolbar` leaving the latter pointing to something that is not displayed and hence has the width 0? Also, as far as I understand, `this.ui.toolbar` here belongs to the `Toolbar` class. Besides this shortcut also the similar shortcut in the `StructureBoard` class in `cms.structureboard.js` might have to be updated. 

It seems that before the commit the DOM **was patched and not replaced** potentially leaving the pointers in tact.

Fix implemented for line 739 in `cms.toolbar.js`:

    this.ui.toolbar.html(newToolbar.children());

This leaves the DOM element unchanged, to which the two `this.ui.toolbar` point.